lib/whitespace.coffee

```diff
@@ -87,3 +97,11 @@ class Whitespace
 
     buffer.transact ->
       buffer.scan new RegExp(spacesText, 'g'), ({replace}) -> replace('\t')
+
+  clearBlankLineOnNewline: (editor, event) ->
+    return unless event.text is "\n"
+    buffer = editor.getBuffer()
+    previousRow = editor.getCursorBufferPosition().row - 1
```

Comment:
Because there might be multiple cursors, we need to determine `previousRow` based on `event` here. The event has a  `range` property, so you should be able to use `range.start`.